src/RegistrationMsg.chpl

```diff
@@ -278,6 +280,116 @@ module RegistrationMsg
         return new MsgTuple(repMsg, MsgType.NORMAL);
     }
 
+    /* 
+    Compile the component parts of a DataFrame attach message 
+
+    :arg cmd: calling command 
+    :type cmd: string 
+
+    :arg payload: name of SymTab element
+    :type payload: string
+
+    :arg argSize: number of arguments in payload
+    :type argSize: int
+
+    :arg st: SymTab to act on
+    :type st: borrowed SymTab 
+
+    :returns: MsgTuple response message
+    */
+    proc attachDataFrameMsg(cmd: string, payload: string, argSize: int,
+                                 st: borrowed SymTab): MsgTuple throws {        
+        var msgArgs = parseMessageArgs(payload, argSize);
+        const name = msgArgs.getValueOf("name"); 
+        var colName = "df_columns_%s".format(name);
+        var repMsg = "dataframe+%s".format(name);
+
+        regLogger.debug(getModuleName(),getRoutineName(),getLineNumber(), 
+                            "%s: Collecting DataFrame components for '%s'".format(cmd, name));
+
+        var jsonParam = msgArgs.get("name");
+        jsonParam.setVal(colName);
+        var json: [0..#1] string = [jsonParam.getJSON()];
+
+        // Add columns as a json list
+        var cols = stringsToJSONMsg(cmd, "%jt".format(json), json.size, st).msg;
+        repMsg += "+json %s".format(cols);
+
+        // Get index 
+        var indParam = new ParameterObj("name", "df_index_%s_key".format(name), ObjectType.VALUE, "");
+        var indJSON: [0..#1] string = [indParam.getJSON()];
```

Comment:
I think you need `asMap` here, not `getJSON`.
```chapel
var indJSON: [0..#1] map(string, string) = [indParam.asMap()];
```

Reply:
@joshmarshall1  - this is my fault. I forgot that `getJSON` is overloaded to support this. In this instance, you are 100% correct because you are creating a new ParameterObj. Resolving this comment.